src/baseroo.test.ts

```diff
@@ -99,3 +99,255 @@ test('convertBase should convert negative float base 16 to 10', () => {
+	// 3.25 = 11.01 in binary = BB.AB in "AB" alphabet
+	const input = '3.25'
+	const actualOutput = convertBase(input, 10, 'AB')
+	const expectedOutput = 'BB.AB'
+	assert.strictEqual(actualOutput, expectedOutput)
+})
+
+test('convertBase should handle precision limits consistently with custom alphabets', () => {
+	// Test that precision is limited to 10 digits for both numeric and custom alphabets
+	// Use 1/3 (0.333...) which has infinite decimal representation
+	const input = '0.3333333333333'
+	const numericOutput = convertBase(input, 10, 2)
+	const customOutput = convertBase(input, 10, 'AB')
+
+	// Both should have the same length (precision limited to 10 fractional digits)
+	const numericFractional = numericOutput.split('.')[1] || ''
+	const customFractional = customOutput.split('.')[1] || ''
+	assert.strictEqual(numericFractional.length, customFractional.length)
+	assert.strictEqual(customFractional.length, 10) // Verify 10-digit precision limit
+})
+
+test('error handling should work properly with custom alphabets', () => {
+	// Test invalid character in custom alphabet
+	assert.throws(() => {
+		convertBase('C', 'AB', 10) // 'C' is not in alphabet 'AB'
+	}, InvalidDigitError)
+
+	// Test custom alphabet with insufficient length
+	assert.throws(() => {
+		convertBase('1', 'A', 10) // Single character alphabet = base 1, which is invalid
+	}, InvalidBaseError)
+
+	// Test invalid character in fractional part with custom alphabet
+	assert.throws(() => {
+		convertBase('A.C', 'AB', 10) // 'C' is not in alphabet 'AB'
+	}, InvalidDigitError)
+})
+
+test('error messages should be descriptive for custom alphabets', () => {
+	// Test that error messages include helpful context
+	try {
+		convertBase('C', 'AB', 10)
+		assert.fail('Should have thrown InvalidDigitError')
+	} catch (error) {
+		assert.ok(error instanceof InvalidDigitError)
+		assert.ok(error.message.includes('C'))
+		assert.ok(error.message.includes('2')) // Base number for 'AB' alphabet
+	}
+
+	try {
+		convertBase('1', 'A', 10)
+		assert.fail('Should have thrown InvalidBaseError')
+	} catch (error) {
+		assert.ok(error instanceof InvalidBaseError)
+		assert.ok(error.message.includes('1')) // Invalid base number
+	}
+})
+
+// Task 4.1: Backward compatibility tests
+test('backward compatibility - all existing number-based operations work unchanged', () => {
+	// Verify that numeric base operations still work exactly as before
+	assert.strictEqual(convertBase('ff', 16, 10), '255')
+	assert.strictEqual(convertBase('1010', 2, 10), '10')
+	assert.strictEqual(convertBase('100', 10, 2), '1100100')
+	assert.strictEqual(convertBase('255', 10, 16), 'ff')
+
+	// Verify floating point operations
+	assert.strictEqual(convertBase('10.5', 10, 2), '1010.1')
+	assert.strictEqual(convertBase('1010.1', 2, 10), '10.5')
+
+	// Verify negative numbers
+	assert.strictEqual(convertBase('-ff', 16, 10), '-255')
+	assert.strictEqual(convertBase('-255', 10, 16), '-ff')
+})
```

Comment:
If these are "backwards compatibility" tests, were they covered in the existing tests? If not, perhaps they should have been.